README.md

```diff
@@ -10,213 +10,302 @@
-export LD_LIBRARY_PATH=`pwd`/open-source/local/lib
+./kvssink_intermittent_sample <stream-name> <testsrc or devicesrc (optional)>
+```
+Setting the source to `testsrc` will use [videotestsrc](https://gstreamer.freedesktop.org/documentation/videotestsrc/?gi-language=c) and to `devicesrc` will use [autovideosrc](https://gstreamer.freedesktop.org/documentation/autodetect/autovideosrc.html?gi-language=c). By default, kvssink uses "DEFAULT_STREAM" as the stream name, and the sample uses videotestsrc as the source. If a KVS stream with the provided or default name does not exist, the stream will automatically be created.
+
+<br>
+
+#### Running Samples in offline mode
+By default, the samples run in near-real-time mode. To set offline mode, set streamInfo.streamCaps.streamingType to `STREAMING_TYPE_OFFLINE`, where, `streamInfo` is of type `StreamInfo`, `streamCaps` is of type `StreamCaps` and `streamingType` is of type `STREAMING_TYPE`.
+
+<br>
+
+### Viewing the Sample Footage
+Check the ingested video stream to confirm successful streaming using either the [AWS KVS Console](https://console.aws.amazon.com/kinesisvideo/home/#/streams/) or the [KVS Media Viewer](https://aws-samples.github.io/amazon-kinesis-video-streams-media-viewer/).
+
+If playback issues are encountered, pleaser refer to the playback requirements under [GetHLSStreamingSessionURL](https://docs.aws.amazon.com/kinesisvideostreams/latest/dg/API_reader_GetHLSStreamingSessionURL.html) or [GetDASHStreamingSessionURL](https://docs.aws.amazon.com/kinesisvideostreams/latest/dg/API_reader_GetDASHStreamingSessionURL.html) and to the [Troubleshooting HLS Issues](https://docs.aws.amazon.com/kinesisvideostreams/latest/dg/hls-playback.html#how-hls-ex1-ts) guide.
+
+<br>
+
+## Build Options
+### Considerations
+- The **kvssink** GStreamer plugin and samples, and the **JNI** are _not_ built by default. To build them, include their corresponding cmake command arguments: `cmake .. -DBUILD_GSTREAMER_PLUGIN=ON -DBUILD_JNI=TRUE`
+- By default, the **dependency libraries** are installed from GitHub and built locally. To instead link to pre-installed libraries on the device, include the following cmake command argument: `cmake .. -DBUILD_DEPENDENCIES=OFF`
+  - The dependency libraries are Curl, OpenSSL, and Log4Cplus.
+ 
+### CMake Arguments
+You can pass the following additional CMake options:
+
+| Option	                     | Default       | Description	|
+|:-----------------------------|:-------------:|:-------------|
+| BUILD_GSTREAMER_PLUGIN       | OFF           | Build the kvssink GStreamer plugin
+| BUILD_JNI                    | OFF           | Build C++ wrapper for JNI to expose the functionality to Java/Android
+| BUILD_DEPENDENCIES           | ON            | Build depending libraries from source
+| BUILD_TEST                   | OFF           | Build unit/integration tests, may be useful to confirm support for your device, to run tests:       `./tst/producerTest`
```

Comment:
Should mention if credentials & region are required to run the tests & whether the resources get cleaned up or not in case of failing tests.

Reply:
I feel this would be too verbose for this table. I will add including a "Running Tests" section to the PR description as a potential future addition where these details would fit in well.

Reply:
I think that's a good idea. Having a section dedicated to how to run the tests and the sanitizers and view the results.